storage/src/cache/worker.rs

```diff
@@ -139,6 +141,13 @@ impl AsyncWorkerMgr {
 
     /// Stop all working threads.
     pub fn stop(&self) {
+        if self
+            .active
+            .compare_exchange(true, false, Ordering::AcqRel, Ordering::Relaxed)
+            .is_err()
+        {
+            return;
```

Comment:
Seems `stop` can't be re-entered without the check if it is active? So we previously had a bug here?

Reply:
Previously, only stop workers when delete image, so stop() will not be re-entered. Anyway we should add active stat to handle the re-enter case now.